src/Pages/ListActivities.php

```diff
@@ -26,93 +29,96 @@ abstract class ListActivities extends Page implements HasForms
+
+    public function mount($record): void
     {
         $this->record = $this->resolveRecord($record);
     }
 
-    public function getBreadcrumb(): string
+    public function getBreadcrumb(): string|Htmlable
     {
         return static::$breadcrumb ?? __('filament-activity-log::activities.breadcrumb');
     }
 
-    public function getTitle(): string
+    public function getTitle(): string|Htmlable
     {
         return __('filament-activity-log::activities.title', ['record' => $this->getRecordTitle()]);
     }
 
     public function getActivities()
     {
         return $this->paginateTableQuery(
-            $this->record->activities()->with('causer')->latest()->getQuery()
+            $this->getActivitiesQuery()
         );
     }
 
+    protected function getActivitiesQuery(): Builder
+    {
+        return $this->record->activities()
+            ->with(['causer' => fn($query) => $query->withTrashed()])
+            ->latest();
+    }
+
     public function getFieldLabel(string $name): string
     {
         static::$fieldLabelMap ??= $this->createFieldLabelMap();
 
-        return static::$fieldLabelMap[$name] ?? $name;
+        return static::$fieldLabelMap->get($name, $name);
     }
 
     protected function createFieldLabelMap(): Collection
     {
-        $form = static::getResource()::form(new Form($this));
+        return $this->extractFormFields()
+            ->filter(fn ($field) => $field instanceof Field)
+            ->mapWithKeys(fn (Field $field) => [
+                $field->getName() => $field->getLabel() ?? $field->getName(),
+            ]);
+    }
 
+    protected function extractFormFields(): Collection
+    {
+        $form = static::getResource()::form(new Form($this));
         $components = collect($form->getComponents());
         $extracted = collect();
 
-        while (($component = $components->shift()) !== null) {
+        while ($component = $components->shift()) {
             if ($component instanceof Field || $component instanceof MorphToSelect) {
                 $extracted->push($component);
-
                 continue;
             }
 
-            $children = $component->getChildComponents();
-
-            if (count($children) > 0) {
-                $components = $components->merge($children);
-
-                continue;
+            if ($children = $component->getChildComponents()) {
+                $components->push(...$children);
             }
-
-            $extracted->push($component);
         }
 
-        return $extracted
-            ->filter(fn ($field) => $field instanceof Field)
-            ->mapWithKeys(fn (Field $field) => [
-                $field->getName() => $field->getLabel(),
-            ]);
+        return $extracted;
     }
 
     public function canRestoreActivity(): bool
     {
         return static::getResource()::canRestore($this->record);
     }
 
-    public function restoreActivity(int|string $key)
+    public function restoreActivity(int|string $key): void
```

Comment:
This could be a breaking change if people overwrite that method, so I'd rather not include it now.